tests/test_typed_dict.py

```diff
@@ -85,3 +87,27 @@ class User(TypedDict):
     }
 
     assert_schema(User, expected)
+
+
+def test_non_total_typed_dict():
+    class Opt(StrEnum):
+        val = "invalid-val"
+
+    class PyType(TypedDict, total=False):
+        name: str
+        age: int | None
+        opt: Opt | None
```

Comment:
might be worth to add a `str | None` here just to verify what happens when we do `Union[str, None, str]`?

Reply:
Done in https://github.com/localstack/py-avro-schema/pull/11/commits/c514ae055c20b12e72bbcfe5fc93f89bef7b3841. `Union` can't have duplicates, so it would just be `Union[str, | None]`. If you actually try to create a type alias like this, you'd get a `Optional[str]` :)